### PR TITLE
GH-2952: Add maven wrapper

### PR DESCRIPTION
**Pull request by nastra**
### Rationale for this change
Having a maven wrapper in the project removes the burden for users to have maven installed on their system.
The wrapper was added by running `mvn wrapper:wrapper`

### CODE REVIEW

**Comment by nastra** [comment]
As a next step (please let me know if that should be within this PR) I was planning to update all `mvn` references in the codebase to `./mvnw`

**Comment by wgtmac** [comment]
This looks nice. However, my concern is the maintenance burden of the new script.

@julienledem @gszadovszky WDYT?

**Comment by joyCurry30** [comment]
Missing mvnw.cmd

File: .mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+wrapperVersion=3.3.2
+distributionType=only-script
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.9.8/apache-maven-3.9.8-bin.zip
```
Comment (joyCurry30) [inline .mvn/wrapper/maven-wrapper.properties:19]: This version is too new and may cause problems.

Comment (Fokko) [reply .mvn/wrapper/maven-wrapper.properties:19]: The CI runs 3.8.8. Did you test this locally @nastra?

Comment (nastra) [reply .mvn/wrapper/maven-wrapper.properties:19]: yes I tested this locally. If there are concerns with that Maven version then I can switch that to a version that is used by CI. Or I could update CI scripts to use the wrapper to make sure that particular maven version runs without issues

**Comment by nastra** [comment]
> This looks nice. However, my concern is the maintenance burden of the new script.
> 
> @julienledem @gszadovszky WDYT?

@wgtmac typically this script doesn't require any maintenance and will be updated automatically when updating the underlying maven version through the wrapper. That doesn't mean that every maven version bump requires an updated script, just that the wrapper task will update the script if necessary

File: .mvn/wrapper/maven-wrapper.properties (hunk 1 above)
Comment (nastra) [reply .mvn/wrapper/maven-wrapper.properties:19]: I've updated CI to run through the wrapper to see if there would be any issues with that version

Comment (Fokko) [reply .mvn/wrapper/maven-wrapper.properties:19]: Nice, that would be my preference. Let's see what others think.

Comment (nastra) [reply .mvn/wrapper/maven-wrapper.properties:19]: @joyCurry30 @Fokko CI seems to be running fine with that Maven version

**Review by Fokko — APPROVED** [review]
I think this is nice, but I'll leave this open for others to chime in as well 👍

**Review by joyCurry30 — APPROVED** [review]

File: mvnw [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# ----------------------------------------------------------------------------
+
+# ----------------------------------------------------------------------------
+# Apache Maven Wrapper startup batch script, version 3.3.2
```
Comment (wgtmac) [inline mvnw:22]: Is it copied from somewhere else? It would be good if a link is provided so we can keep it updated.

Comment (nastra) [reply mvnw:22]: no, it's being generated and will be automatically updated (if necessary) when updating the underlying maven version of the wrapper

Comment (nastra) [reply mvnw:22]: if you run `mvn wrapper:wrapper` against `master` you'll get the same scripts generated

**Review by wgtmac — APPROVED** [review]
LGTM

Thanks @nastra!

**Pull request merged**